src/azure-cli/azure/cli/command_modules/appservice/custom.py

```diff
@@ -1353,6 +1353,157 @@ def get_webapp_sitecontainer_log(cmd, name, resource_group, container_name, slot
         raise AzureInternalError("Failed to fetch sitecontainer logs. Error: {}".format(str(ex)))
 
 
+def convert_webapp_sitecontainers(cmd, name, resource_group, mode, slot=None):
+    """
+    Convert a webapp between classic and sitecontainers mode.
+
+    :param cmd: CLI command context
+    :param name: Name of the webapp
+    :param resource_group: Resource group of the webapp
+    :param mode: Target mode, either 'classic' or 'sitecontainers'
+    :param slot: Optional deployment slot
+    """
+    client = web_client_factory(cmd.cli_ctx)
+    if mode not in ['classic', 'sitecontainers']:
+        raise InvalidArgumentValueError(
+            "Invalid mode '{}'. Allowed values: classic, sitecontainers.".format(mode)
+        )
+
+    site_config = get_site_configs(cmd, resource_group, name, slot)
+    linux_fx_version = getattr(site_config, "linux_fx_version", None)
+ 
+    if mode == 'sitecontainers':
+        if linux_fx_version and not linux_fx_version.startswith('DOCKER|'):
+            raise ValidationError("Cannot convert to sitecontainers mode as site is not a classic custom container app.")
+        acr_use_managed_identity_creds = getattr(site_config, "acr_use_managed_identity_creds", None)
+        acr_user_managed_identity_id = getattr(site_config, "acr_user_managed_identity_id", None)
+
+        acr_user_name = None
+        acr_user_password = None
+
+        from azure.cli.core.commands.client_factory import get_subscription_id
+        subscription_id = get_subscription_id(cmd.cli_ctx)
+        url = (
+            f"/subscriptions/{subscription_id}/resourceGroups/{resource_group}/"
+            f"providers/Microsoft.Web/sites/{name}/config/appsettings/list?api-version=2023-12-01"
+            )
+        request_url = cmd.cli_ctx.cloud.endpoints.resource_manager + url
+        response = send_raw_request(cmd.cli_ctx, "POST", request_url)
+        app_settings_raw = response.json()
+        app_settings = app_settings_raw.get("properties", {})
+
+        acr_user_password = app_settings.get("DOCKER_REGISTRY_SERVER_PASSWORD", None)
+        acr_user_name = app_settings.get("DOCKER_REGISTRY_SERVER_USERNAME", None)
+        websites_port = app_settings.get("WEBSITES_PORT", None) or app_settings.get("PORT", None)
+
+        # Get docker image from linux_fx_version
+        docker_image = linux_fx_version.replace("DOCKER|", "", 1)
+        sidecarmainname = "main"
```

Comment:
could we follow the same convention in naming variables ?
sidecar_maincontainer_name